lib/ci-helper.ts

```diff
@@ -763,6 +772,211 @@ export class CIHelper {
+                        throw new Error("Request author is not commit author.");
+                    }
+                });
+            }
+
+            const files = prInfo.files;
+            if (files && files.totalCount && files.nodes) {
+                files.nodes.map(node => {
+                    const changedFile = node as PullRequestChangedFile;
+                    this.checkNewUserFile(changedFile, files.totalCount,
+                        genLabel);
+                });
+            } else {
+                throw new Error("No files updated.");
+            }
+
+            const queryRef = `query {
+                repository(name: "${pr.baseRepo}", owner: "${pr.author}") {
+                    ref(qualifiedName: "${prInfo.baseRefName}") {
+                        name
+                    }
+                }
+            }`;
+
+            const refInfo = await this.github.graphql<queryResponse>(
+                pr.baseOwner,
+                queryRef
+            );
+
+            if (!refInfo || !refInfo.repository.ref) {
+                throw new Error(`Base branch ${prInfo.baseRefName
+                    } not found in ${pr.author}/${pr.baseRepo}.`);
+            }
+        } catch (e) {
+            if ((e as labelInfo).label === undefined ) {
+                const error = e as Error;
+                result = false;
+                await this.github.closePR(prInfo.url, `PR is being closed:\n\n${
+                    error.message}`);
+            } else {
+                const label = e as labelInfo;
+                await this.github.addPRLabels(prInfo.url, [label.label]);
+                await this.github.addPRComment(prInfo.url, `Label \`${
+                    label.label}\` added for this reason:\n\n${
+                    label.message}`);
+            }
+        }
+
+        return result;
+    }
+
+    /**
+     * Perform some validation tests on a file in a pull request.
+     *
+     * @param changedFile file that is part of a commit in a pull request
+     * @param files number of files that are changed in the pull request
+     */
+    public checkNewUserFile(changedFile: PullRequestChangedFile,
+        files: number, setLabel: setLabelInfo):
+        void {
+        const filenameParts = path.parse(changedFile.path);
+
+        // one file with only deletions?
+        if (files === 1 &&
+            !changedFile.additions && changedFile.deletions) {
+            throw setLabel("validate", `File \`${changedFile.path
+                }\` expecting more than one change.`)
+        }
+
+        if (filenameParts.name.match(/.+\./)) {
+            throw new Error(`File \`${changedFile.path
+                }\` not allowed with multi-part name`);
+        }
+
+        if (filenameParts.name.match(/ /)) {
+            throw new Error(`File \`${changedFile.path
+                }\` not allowed with space in name`);
+        }
+
+        const badNames = ["SECURITY.md", "README"];
```

Comment:
Can we tighten this `SECURITY.md` test? As I mentioned elsewhere, I would _love_ for us to have a `SECURITY.md`, it's just a lot of work to put in the relevant information. Looking at an example from https://github.com/git/git/pulls?q=is%3Apr+security.md+is%3Aclosed, I see that this would be a good needle:

> Use this section to tell people about

If the PR contains a `SECURITY.md` file containing these words, it is _so_ clearly not a quality contribution that we can close it straight away.

Reply:
Was hoping to avoid checking file content.  Some of the bogus PRs had files with names containing `SECURITY.md`.  Since there is not a `SECURITY.md` at the moment, it seems unlikely that a first time contributor would be providing a complete one and this check only adds a label to the PR.

Reply:
As I said a couple of times, I think it would make sense for Git to have a `SECURITY.md` because it is shown prominently in GitHub's [`Security` tab](https://github.com/git/git/security/policy) (if there _is_ a `SECURITY.md`). But it will take quite a bit more effort to craft a good one, the contributions we got so far were really inadequate: they essentially left the template alone that was supposed to be adapted to whatever project the contribution was supposed to target.

> Was hoping to avoid checking file content.

Ah. Maybe a good indication that we do need the commits locally, and that we do need a real parser so that we can perform all the checks that we are interested in?